File: C Sharp/Object Validator/Runtime/Length.cs
using System.Collections;

namespace Quicksilver.ObjectValidator.Runtime;
class Length(bool reversed, string? fieldExpression, string range) : Condition(reversed, fieldExpression)
{
    private readonly string range = range;

    protected override bool IsFulfilledBy(object? value, string? fullFieldExpression, HashSet<string> passedFields, HashSet<string> failedFields)
    {
        return value switch {
            null => throw new Exception("Null values are not supported for 'length'."),
            string s => Utils.InRange(s.Length, range),
            ICollection c => Utils.InRange(c.Count, range),
            IReadOnlyCollection<object> roc => Utils.InRange(roc.Count, range),
            _ => throw new Exception("Unsupported type for 'length': " + value.GetType())
        };
    }
}

File: Java/Object Validator/lib/src/main/java/com/quicksilver/objectvalidator/runtime/Length.java
package com.quicksilver.objectvalidator.runtime;

import java.util.Collection;
import java.util.HashSet;
import java.util.Map;

public class Length extends Condition {
    private final String range;

    public Length(boolean reversed, String fieldExpression, String range) {
        super(reversed, fieldExpression);
        this.range = range;
    }

    @Override
    protected boolean isFulfilledBy(Object value, String fullFieldExpression, HashSet<String> passedFields, HashSet<String> failedFields) {
        return switch (value) {
            case null -> throw new RuntimeException("Null values are not supported for 'length'.");
            case CharSequence s -> Utils.inRange(s.length(), range);
            case Object[] a -> Utils.inRange(a.length, range);
            case Collection<?> c -> Utils.inRange(c.size(), range);
            case Map<?, ?> m -> Utils.inRange(m.size(), range);
            default -> throw new RuntimeException("Unsupported type for 'length': " + value.getClass());
        };
    }
}